### PR TITLE
Annotate remaining code with _Must_inspect_result_

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Signed-off-by: Alan Jowett <alan.jowett@microsoft.com>

Partially resolves: https://github.com/microsoft/ebpf-for-windows/issues/1542

## Description

Annotate remaining code with _Must_inspect_result_.

## Testing

CI/CD

## Documentation

No.

### CODE REVIEW

File: include/kernel/ebpf_registry_helper.h [hunk 1]
```diff
@@ -6,12 +6,14 @@
...
 
 #define __return_type NTSTATUS
-#define IS_SUCCESS(x) (NT_SUCCESS(x))
 #define _SUCCESS STATUS_SUCCESS
+#define IS_SUCCESS(x) (NT_SUCCESS(x))
 
 #define REG_CREATE_FLAGS 0
 #define GUID_STRING_LENGTH 38 // not including the null terminator.
 
+typedef _Return_type_success_(NT_SUCCESS(return )) uint32_t ebpf_registry_result_t;
```
Comment (dthaler) [inline include/kernel/ebpf_registry_helper.h:15]: bug
```suggestion
typedef _Return_type_success_(NT_SUCCESS(return == 0)) uint32_t ebpf_registry_result_t;
```

Comment (Alan-Jowett) [reply include/kernel/ebpf_registry_helper.h:15]: Not sure I follow. NT_SUCCESS is defined as a comparison of >= 0:
```
#define NT_SUCCESS(Status)  (((NTSTATUS)(Status)) >= 0)
```

**Review by dthaler — DISMISSED** [review]

File: libs/api_common/store_helper_internal.cpp [hunk 2]
```diff
@@ -139,7 +139,8 @@ _load_program_data_information(
 
         // Read "is_privileged".
         status = read_registry_value_dword(program_info_key, EBPF_PROGRAM_DATA_PRIVELEGED, &is_privileged);
-        if (result != EBPF_SUCCESS) {
+        if (status != EBPF_SUCCESS) {
```
Comment (saxena-anurag) [inline libs/api_common/store_helper_internal.cpp:142]:
```suggestion
        if (status != ERROR_SUCCESS) {
```

**Review by saxena-anurag — DISMISSED** [review]

**Comment by dthaler** [comment]
@Alan-Jowett analyze/build failed with
" D:\a\ebpf-for-windows\ebpf-for-windows\libs\api_common\store_helper_internal.cpp(141): error C28193: 'status' holds a value that must be examined. [D:\a\ebpf-for-windows\ebpf-for-windows\libs\api_common\api_common.vcxproj]"

**Review by dthaler — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]

**Pull request merged**